File: SWEA/Brute_Force/D2/1974. 스도쿠 검증.py
T = int(input())

for tc in range(1, T + 1):
    sudoku = []
    for i in range(9):
        sudoku.append(list(map(int, input().split())))

    def check_row(x):
        check_list = []
        for i in range(9):
            if sudoku[x][i] in check_list:
                return 0
            check_list.append(sudoku[x][i])
        return 1

    def check_col(y):
        check_list = []
        for i in range(9):
            if sudoku[i][y] in check_list:
                return 0
            check_list.append(sudoku[i][y])
        return 1

    def check_square(x, y):
        check_list = []
        for i in range(3):
            for j in range(3):
                if sudoku[x + i][y + j] in check_list:
                    return 0
                check_list.append(sudoku[x + i][y + j])
        return 1

    def check_3b3():
        for i in [0, 3, 6]:
            for j in [0, 3, 6]:
                if not check_square(i, j):
                    return 0
        return 1

    def check():
        for i in range(9):
            if not check_row(i):
                return 0
            if not check_col(i):
                return 0
        if not check_3b3():
            return 0
        return 1
    print("#", end="")
    print(tc, check())

File: SWEA/Brute_Force/D2/1974. 스도쿠 검증.java
import java.sql.SQLOutput;
import java.util.*;
import java.io.*;
public class Main {
    public static int check(int sudoku[][])
    {
        for (int i = 0; i < 9; i++) {
            if (check_row(sudoku, i) == 0)
                return 0;
            if (check_col(sudoku, i) == 0)
                return 0;
        }
        for (int i = 0; i < 9; i+=3)
            for (int j = 0; j < 9; j+=3) {
                if (check_3b3(sudoku, i, j) == 0)
                    return 0;
            }
        return 1;
    }
    public static int check_row(int[][] sudoku, int x)
    {
        ArrayList checkList = new ArrayList();
        for (int i = 0; i < 9; i++){
            if (checkList.contains(sudoku[x][i]))
                return 0;
            checkList.add(sudoku[x][i]);
        }
        return 1;
    }
    public static int check_col(int[][] sudoku, int y)
    {
        ArrayList checkList = new ArrayList();
        for (int i = 0; i < 9; i++){
            if (checkList.contains(sudoku[i][y]))
                return 0;
            checkList.add(sudoku[i][y]);
        }
        return 1;
    }
    public static int check_3b3(int[][] sudoku, int x, int y)
    {
        ArrayList checkList = new ArrayList();
        for (int i = 0; i < 3; i++){
            int nx = x + i;
            for (int j = 0; j < 3; j++){
                int ny = y + j;
                if (checkList.contains(sudoku[nx][ny]))
                    return 0;
                checkList.add(sudoku[nx][ny]);
            }
        }
        return 1;
    }
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        for (int tc = 1; tc <= T; tc++) {
            int sudoku[][] = new int[9][9];
            for (int i = 0; i < 9; i++)
                for (int j = 0; j < 9; j++)
                    sudoku[i][j] = sc.nextInt();
//            System.out.println(Arrays.deepToString(sudoku));
            System.out.println("#" + tc + " " + check(sudoku));
        }
    }
}
